### PR TITLE
Upgrade to Legolas 0.5

### LINKED ISSUE #96: upgrade to Legolas 0.5
We define two schemas
- `"lighthouse.evaluation@1",`
- `"lighthouse.observation@1"`
- `"lighthouse.class@1"`
- `"lighthouse.label-metrics@1" > "lighthouse.class@1"`
- `"lighthouse.hardened-metrics@1" >"lighthouse.class@1"`
- `"lighthouse.tradeoff-metrics@1" >"lighthouse.class@1"`

We use `Legolas.validate` in code and tests.

## TODOs

- [ ] upgrade evaluation schema
- [ ] upgrade observation schema
- [ ] upgrade class schema
- [ ] upgrade label-metrics schema
- [ ] upgrade hardened-metrics schema
- [ ] upgrade tradeoff-metrics schema

### PR DESCRIPTION
**Pull request by a-cakir**
Closes #96

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [deps]
 CairoMakie = "13f3f980-e62b-5c42-98c6-ff1f3baf88f0"
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+Lighthouse = "ac2c24cd-07f0-4848-96b2-1b82c3ea0e59"
```
Comment (a-cakir) [inline docs/Project.toml:4]: Am I missing something or this really shouldnt be here?

File: src/row.jl [hunk 2]
```diff
@@ -1,162 +1,133 @@
...
+                                                n::Int64}}}
+    per_class_pr_curves::Union{Missing,Vector{GenericCurve}}
+    per_class_reliability_calibration_curves::Union{Missing,Vector{GenericCurve}}
+    per_class_reliability_calibration_scores::Union{Missing,Vector{Float64}}
+    per_class_roc_aucs::Union{Missing,Vector{Float64}}
+    per_class_roc_curves::Union{Missing,Vector{GenericCurve}}
+    per_expert_discrimination_calibration_curves::Union{Missing,Vector{GenericCurve}}
+    per_expert_discrimination_calibration_scores::Union{Missing,Vector{Float64}}
+    spearman_correlation::Union{Missing,
+                                @NamedTuple{ρ::Float64,  # Note: is rho not 'p' 😢
```
Comment (hannahilea) [inline src/row.jl:40]: LOL :ohno:

File: src/row.jl [hunk 3]
```diff
@@ -188,7 +159,7 @@ function _inputs_to_observation_table(; predicted_hard_labels::AbstractVector,
                                       elected_hard_labels::AbstractVector,
                                       votes::Union{Nothing,Missing,AbstractMatrix}=nothing)
     votes_itr = has_value(votes) ? eachrow(votes) :
-                (missing for _ in 1:length(predicted_hard_labels))
+                Iterators.repeated(missing, length(predicted_hard_labels))
```
Comment (hannahilea) [inline src/row.jl:162]: til

File: src/metrics.jl [hunk 4]
```diff
@@ -340,21 +340,21 @@ end
...
+    return _evaluation_dict(row)
 end
 
 """
-    evaluation_metrics_row(observation_table, classes, thresholds=0.0:0.01:1.0;
+    evaluation_metrics_record(observation_table, classes, thresholds=0.0:0.01:1.0;
                            strata::Union{Nothing,AbstractVector{Set{T}} where T}=nothing,
                            optimal_threshold_class::Union{Missing,Nothing,Integer}=missing)
-    evaluation_metrics_row(predicted_hard_labels::AbstractVector,
+    evaluation_metrics_record(predicted_hard_labels::AbstractVector,
```
Comment (kleinschmidt) [inline src/metrics.jl:357]: even though this isn't exported, it is documented, so it seems like a good idea to provide a deprecation path (at least throw an error that points to the new method).

**Review by kleinschmidt — APPROVED** [review]
looks great!! I took the liberty of formatting, and adding a formatting step to future CI (there were a bunch of places where the indentation got mangled).

File: src/deprecations.jl [hunk 5]
```diff
@@ -0,0 +1,4 @@
+function evaluation_metrics_row(args...; kwargs...)
+    error("`Lighthouse.evaluation_metrics_row` has been removed in favor of " *
+          "`Lighthouse.evaluation_metrics_record`.")
```
[the remark this answers is not in the document]
Comment (kleinschmidt) [reply src/deprecations.jl:3]: ah, the consequences of my own decisions coming back to bite me I see

**Pull request merged**